generators/generator-microsoft-bot-adaptive/generators/app/templates/dotnet/functions/Triggers/MessagesTrigger.cs

```diff
@@ -34,21 +47,35 @@ public MessagesTrigger(IBotFrameworkHttpAdapter adapter, IBot bot)
         /// The <see cref="IActionResult"/>.
         /// </returns>
         [FunctionName("messages")]
-        public async Task<IActionResult> RunAsync([HttpTrigger(AuthorizationLevel.Anonymous, "get", "post", Route = null)] HttpRequest req)
+        public async Task<IActionResult> RunAsync([HttpTrigger(AuthorizationLevel.Anonymous, "get", "post", Route = "{adapter}")] HttpRequest req, string adapter)
         {
-            // Delegate the processing of the HTTP POST to the adapter.
-            // The adapter will invoke the bot.
-            await _adapter.ProcessAsync(req, req.HttpContext.Response, _bot).ConfigureAwait(false);
-
-            if (req.HttpContext.Response.IsSuccessStatusCode())
+            if (string.IsNullOrEmpty(route))
             {
-                return new OkResult();
+                throw new ArgumentNullException(nameof(route));
             }
 
-            return new ContentResult()
+            IBotFrameworkHttpAdapter adapter;
+
+            if (_adapters.TryGetValue(route, out adapter))
             {
-                StatusCode = req.HttpContext.Response.StatusCode,
-            };
+                // Delegate the processing of the HTTP POST to the appropriate adapter.
+                // The adapter will invoke the bot.
+                await adapter.ProcessAsync(req, req.HttpContext.Response, _bot).ConfigureAwait(false);
+
+                if (req.HttpContext.Response.IsSuccessStatusCode())
+                {
+                    return new OkResult();
+                }
+                
+                return new ContentResult()
+                {
+                    StatusCode = req.HttpContext.Response.StatusCode,
+                };
+            }
+            else
```

Comment:
I think we can remove the else statement since we will always return if we hit true on the above condition, and just move the KeyNotFoundException up a scope.